flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -88,6 +104,12 @@
 
 	private final boolean enableCreditBased;
 
+	private final boolean enableNetworkDetailedMetrics;
+
+	private final ConcurrentHashMap<String, ResultPartition[]> allPartitions;
```

Comment:
resultPartitionManager already has "partition by execution id" mapping.
why do we need to introduce extra `taskId`? task seems to be identifiable by execution id.

Reply:
Actually there is another consideration for this issue. 
Currently we pass both `taskId` and `executionId` in the method of `ShuffleService#createResultPartitionWriters`. The parameters seem a bit duplicated  and not clean. The `taskId` is used internally in `ResultPartition` for debugging log and it is more readable than `executionId`. The `executionId` parameter could be removed future from this method if it is contained directly from `ResultPartitionDeploymentDescriptor` which is proposed in #7835 . So I used `taskId` for index here for that consideration.
If we do not need `taskId` for debugging info in `ResultPartition`, then we can remove this parameter and only retain `executionId` for index. Or we retain both parameters as now here and change to use `executionId` for index, then in future we could consider how to handle `taskId` to make the interface simple?

Reply:
At the first glance, it looks like we could include task name into `ResultPartitionDeploymentDescriptor` as well in #7835 along with `executionId`.
I would still index by `executionId` if needed (add to method at the moment before #7835), because it is more reliable and concise than task name + id used for debug.

Reply:
Sorry for missing this comment before.

Yes, I agree with the point of indexing by `executionId`, and use task name + id for debugging.
My current thought is passing `ShuffleTaskInfo` into `ShuffleService` during creating partition/gate, so #7835 would not be used any more. I would double check the details from the implementation.